Code/MinimalLib/minilib.cpp

```diff
@@ -73,32 +139,26 @@ ROMol *qmol_from_input(const std::string &input) {
 }
 
 std::string svg_(const ROMol &m, unsigned int w, unsigned int h,
-                 const std::vector<unsigned int> *atomIds = nullptr,
-                 const std::vector<unsigned int> *bondIds = nullptr) {
-  MolDraw2DSVG drawer(w, h);
-  std::vector<int> *highlight_atoms = nullptr;
-  if (atomIds && atomIds->size()) {
-    highlight_atoms = new std::vector<int>;
-    highlight_atoms->reserve(atomIds->size());
-    for (auto ai : *atomIds) {
-      highlight_atoms->push_back(ai);
+                 const std::string &details = "") {
+  std::vector<int> atomIds;
+  std::vector<int> bondIds;
+  std::string legend = "";
+  if (!details.empty()) {
+    int offsetx = 0, offsety = 0;
+    auto problems = process_details(details, w, h, offsetx, offsety, legend,
```

Comment:
I feel there might be a better way to return either a text string OR an svg.

Perhaps a helper function?  

```
bool is_problem(const std::string &svg_string)
```

at the minimum...

Reply:
The `is_problem()` thing is something that client code could do pretty easily if they were concerned about that. I'd prefer to avoid adding mandatory overhead/complications to this code.

Figuring out a general-purpose way of handling/reporting problems in these functions is an interesting idea. If we want to assume single-threaded use of these functions it's easy to add a "last_operation_succeeded" flag; seems tricky otherwise. In any case, that would be for a different PR

Reply:
Perhaps we could return a blank or svg with a broken molecule symbol with the error in the desc field:


```
<svg ...>
<g>
<desc>
  ERROR: JSON contains 'offsetx' field, but it is not an int
<desc>
...
</g>
</svg>
```

Not the prettiest, but at least won't break any workflows and allows javascript to probe the "desc" tag to get errors.

Reply:
at the moment it's even easier and faster than that: if you want to check whether nor not mol.get_svg() succeeded all you have to do is check to see whether or not it starts with `<?xml`. Anything else indicates failure and can be handled.
If we actually return SVG with error messages, then that will need to be parsed every time in order to determine whether or not there was an error.